### DISCUSSION TITLE
Massive assignment of updated data

### DISCUSSION THREAD

**Discussion by robertklein** [prisma/prisma#4445] [Q&A] [answered]
I have a table `categories` where columns `level` and `is_active` are NOT NULL.
```sql
CREATE TABLE "public"."category" (
  "id" int4 NOT NULL DEFAULT nextval('category_id_seq'::regclass),
  "level" int4 NOT NULL,
  "sort" int4,
  "is_active" int4 NOT NULL DEFAULT 1,
  "deleted_at" timestamp(3)
)
;
```

nexus input object type:
```js
export const categoryUpdateInput = inputObjectType({
  name: 'categoryUpdateInput',
  definition(t) {
    t.int('sort')
    t.int('level', { default: undefined })
    t.int('is_active', { default: undefined })
    t.string('deleted_at')
  },
})
```
nexus mutation:
```js
t.field('updateCategory', {
  type: category,
  args: {
    id: nonNull(intArg()),
    values: nonNull(categoryUpdateInput)
  },
  resolve: async (_, { id, values }, ctx) => {        
    return await ctx.prisma.category.update({
      where: {
        id: id
      },
      data: values
    })
```

I want to set all the attributes at once and avoid of doing something like:
```sql
t.field('updateCategory', {
  type: category,
  args: {
    id: nonNull(intArg()),
    values: nonNull(categoryUpdateInput)
  },
  resolve: async (_, { id, values }, ctx) => {        
    return await ctx.prisma.category.update({
      where: {
        id: id
      },
      data: {
        sort: values.sort,
        level: values.level ? values.level : undefined,
        is_active: values.is_active ? values.is_active : undefined,
        deleted_at: values.deleted_at
      }
    })
```

Is there any way how to do it?

In case of assignment `data: values` I have a ts error:
```
Types of property 'level' are incompatible.
  Type 'number | null | undefined' is not assignable to type 'number | IntFieldUpdateOperationsInput | undefined'.
    Type 'null' is not assignable to type 'number | IntFieldUpdateOperationsInput | undefined'.
```

**Comment by ryands17** [comment]
Hey @robertklein 👋 

The only way in this case would be to create an object by stripping out the empty values (`null` and `undefined`) and pass that object to `data` in the `update` method.

**Reply by robertklein** [reply]
But how can I save `{ sort: null }` (it is legal value for NULL column in db)? It would be stripped out too.

**Comment by robertklein** [comment, marked as answer]
`values as Prisma.categoryUpdateInput` solves my problem

```js
t.field('updateCategory', {
  type: category,
  args: {
    id: nonNull(intArg()),
    values: nonNull(categoryUpdateInput)
  },
  resolve: async (_, { id, values }, ctx) => {        
    return await ctx.prisma.category.update({
      where: {
        id: id
      },
      data: values as Prisma.categoryUpdateInput
    })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
